### DISCUSSION TITLE
Custom window sizes (height/width)

### DISCUSSION THREAD

**Discussion by pryley** [rxhanson/Rectangle#362] [Ideas]
What about an option to add custom window sizes? This would be useful when taking application screenshots.

I guess this idea is similar to this: https://github.com/rxhanson/Rectangle/discussions/291

**Comment by rxhanson** [comment]
I don't have plans to add this in, but would consider a pull request that doesn't modify the UI. Meaning you could configure it entirely with a terminal command - keyboard shortcut (keycode/modifiers), dimensions, and origin point.

I did build this into [Hookshot](https://hookshot.app) (my paid app). Here's a post I wrote on how to use it: https://medium.com/ryan-hanson/custom-window-dimension-shortcuts-in-macos-e7df347dfd24

**Reply by pryley** [reply]
Awesome! I didn't know that Rectangle had an older sibling.

**Comment by tom79** [comment]
I would also really like this feature. It's available in BetterTouchTool which is functionally excellent but hideous to use and configure. It looks like something that was designed in the 80s.

If you could create custom keyboard shortcuts that moved the current window to coordinates X,Y while also (optionally) making it resize to W,H then I would be very happy indeed! Why? As an example, if you have a large monitor you don't necessarily want to make one window fill the whole screen but you do want to resize it and also center it with one keyboard shortcut. I also use this function a lot for the finder windows which end up being all shapes and sizes and in different positions. I prefer to have the finder window always open with the same size and in the same location.

**Reply by rxhanson** [reply]
Just to clarify, are you looking for something that is basically grabbing a window's existing size/position and reusing that size/position when a shortcut is executed? (Rather than manually specifying dimensions)

**Comment by tom79** [comment]
Ideally you could specify the dimensions you want, so changing the current size AND then also moving it to a particular location where you’d enter in either coordinates or a % of the pixels you have. For example, I like my finder/folder windows to be slightly above the vertical center but horizontally centered. Where as for an app like mail or whatever it’s good to have a larger size window but centered. So with one shortcut (that you could set) it would either resize the window OR move the window OR both. For example, I have Ctrl + Alt + M to center the window while setting it to a specified size and I have Ctrl + Alt + F for finder/folder windows which are not so large and slightly above center as is the macOS default.

Maybe I have a little OCD but the thing I don’t like about macOS is how windows are forever opening in different sizes and in different locations and this is the best way I’ve found to try and control the randomness!

**Reply by rxhanson** [reply]
If I understand correctly, then what you are looking for does exist within the custom shortcut builder in my paid superset of Rectangle, [Hookshot](https://hookshot.app) (as mentioned above, along with a [post](https://medium.com/ryan-hanson/custom-window-dimension-shortcuts-in-macos-e7df347dfd24) about how to use it and why this is not in Rectangle). If I am not understanding, let me know.

**Comment by tom79** [comment]
Hookshot is brilliant! How did I not know about this? You need to get it on the App Store. It’s like a more fully featured version of Magnet which is number 4 in the productivity charts and does only a fraction of what Hookshot does.

If I was being nitpicky, it is SO close to replicating the functionality I was describing. If you create a custom shortcut and select “Custom” for the docking position and use percentages for the X and Y coordinates, it seems to always use the bottom left coordinate of the window. If you wanted to, for example, resize the window to 800 x 600 and then position the *center* of the window at coordinates X, Y there is no way to do so. I can select “Center” for the docking position instead of Custom but then the window is centered in the center of the screen, not at the location I want. An easy fix might be to allow the user to select which coordinate is used to perform the calculation in “Custom” mode : Top Left, Bottom Left, Top Right, Bottom Right *and in my case* Center. So in my particular example, I would choose to use the Custom docking position option, but then select the Center option for which coordinate to use.

Otherwise, Hookshot appears to be excellent and is what I’ve been looking for. I lose some other functions like setting function keys to play music, stop music, change volume and I also lose mouse button functions (I have the middle mouse button set to Misson Control for example) but perhaps there is another app to handle that side of things I’m not sure. I just want to get away from Better Touch Tool because the UI is so poorly designed.

**Reply by rxhanson** [reply]
Thanks for the feedback! 

Apple no longer allows apps like Rectangle or Hookshot on the App Store. 

I see what you mean on the coordinates. I have plans to add some behavior for adjusting windows in the Hookshot custom shortcut builder with respect to the window itself, rather than the screen, so this might get solved there.
